eeg_flow/tasks/epochs_evoked.py

```diff
@@ -0,0 +1,434 @@
+    #                  # unit: V (EOG channels)
+    #               )
+
+    stim_before = [el[2] for el in epochs.events]
+    count_stim_before = Counter(stim_before)
+
+
+
+
+    epochs.drop_bad(reject=reject)
+
+    stim_after = [el[2] for el in epochs.events]
+    count_stim_after = Counter(stim_after)
+
+    data = [['1', count_stim_before[1]-count_stim_after[1]], 
+            ['2', count_stim_before[2]-count_stim_after[2]], 
+            ['3', count_stim_before[3]-count_stim_after[3]]] 
+
+    df_count = pd.DataFrame(data, columns=['Stim', 'nb_dropped']) 
+
+    df_count.to_csv(DERIVATIVES_SUBFOLDER / (FNAME_STEM + "_step9_c2-drop-epochs.csv"))
+
+    fig = epochs.plot_drop_log(subject=FNAME_STEM)
+
+    FNAME_DROP_LOG = (
+        DERIVATIVES_SUBFOLDER / "plots" / (FNAME_STEM + "_step9_epochs-rejected.svg")
+    )
+    fig.savefig(FNAME_DROP_LOG, transparent=True)
+
+    totals = Counter(i for i in list(itertools.chain.from_iterable(epochs.drop_log)))
+    df_drops = pd.DataFrame.from_dict(totals, orient = "index")
+    df_drops = df_drops.rename(columns={0: FNAME_STEM})
+    df_drops = df_drops.sort_values(by=[FNAME_STEM], ascending = False)
+    df_drops.to_csv(DERIVATIVES_SUBFOLDER / (FNAME_STEM + "_step9_c3-drop-channel-log.csv"))
+
+    return(epochs)
+
+def save_evoked(epochs, event_id, FNAME_STEM, DERIVATIVES_SUBFOLDER):
+
+    epochs.metadata.groupby(
+        by=[
+            "event_name",
+            "response_correct",
+        ]
+    ).count()
+    # this keeps correct responses only (hits and correct rejection)
+    epochs["response_correct"]
+
+    # %%
+    all_evokeds = dict(
+        (cond, epochs["response_correct"][cond].average()) for cond in event_id
+    )
+    # all_evokeds = {cond: epochs["response_correct"][cond].average() for cond in event_id}
+    all_evokeds
+
+    FNAME_EV_STANDARD = DERIVATIVES_SUBFOLDER / (
+        FNAME_STEM + "_step9_d1-standard_evoked-ave.fif"
+    )
+    FNAME_EV_TARGET = DERIVATIVES_SUBFOLDER / (
+        FNAME_STEM + "_step9_d2-target_evoked-ave.fif"
+    )
+    FNAME_EV_NOVEL = DERIVATIVES_SUBFOLDER / (
+        FNAME_STEM + "_step9_d3-novel_evoked-ave.fif"
+    )
+
+    all_evokeds["standard"].save(FNAME_EV_STANDARD)
+    all_evokeds["target"].save(FNAME_EV_TARGET)
+    all_evokeds["novel"].save(FNAME_EV_NOVEL)
+
+
+def SDT2(hits, misses, fas, crs):
+    """Return a dict with d-prime measures.
+    Parameters
+    ----------
+    %(hits)s
+    %(misses)s
+    %(fas)s
+    %(crs)s
+    """
+    Z = norm.ppf
+
+    # Calculate hit_rate and avoid d' infinity
+    hit_rate = hits / (hits + misses)
+
+    # Calculate false alarm rate and avoid d' infinity
+    fa_rate = fas / (fas + crs)
+
+    # Return d', beta, c and Ad'
+    out = {}
+    out["d"] = Z(hit_rate) - Z(fa_rate)
+    out["beta"] = math.exp((Z(fa_rate) ** 2 - Z(hit_rate) ** 2) / 2)
+    out["c"] = -(Z(hit_rate) + Z(fa_rate)) / 2
+    out["Ad"] = norm.cdf(out["d"] / math.sqrt(2))
+
+    return out
+
+
+
```

Comment:
2 empty-lines between functions.
Simply fixed by setting `black` as your default formater and by formatting your files with `black` before pushing.
For instance in a console: `black path/to/py-files`